pkg/apis/events/v1alpha1/cloudstoragesource_validation.go

```diff
@@ -72,6 +80,18 @@ func validateSecret(secret *corev1.SecretKeySelector) *apis.FieldError {
 	return errs
 }
 
+func validateGCPServiceAccount(gServiceAccountName *string) *apis.FieldError {
+	pattern := `[A-Za-z0-9._%+-]+@[A-Za-z0-9.-]+\.iam.gserviceaccount.com`
```

Comment:
will be better if you compile this once in a var, and then use it. See an example here: https://github.com/knative/eventing/pull/2503/files

Start the regex with ^ and finish it with $

Did you take this service account from some Google documentation? If so, then add the link for future reference... If not, are we sure if can contain `%` and `-` before the `@`, but not after?